x/valset-pref/validator_set.go

```diff
@@ -17,19 +16,61 @@ type valSet struct {
 	amount  sdk.Dec
 }
 
-func (k Keeper) SetValidatorSetPreference(ctx sdk.Context, delegator string, preferences []types.ValidatorPreference) ([]types.ValidatorPreference, error) {
-	// check if a user already has a validator-set created
-	existingValidators, found := k.GetValidatorSetPreference(ctx, delegator)
-	if found {
-		// check if the new preferences is the same as the existing preferences
-		isEqual := k.IsValidatorSetEqual(preferences, existingValidators.Preferences)
-		if isEqual {
-			return nil, fmt.Errorf("The preferences (validator and weights) are the same")
+// SetValidatorSetPreference creates or updates delegators validator set.
+// 1. if valset doesnot exist & user doesnot have existing delegation, store new valset.
+// 2. if valset doesnot exist, but user has existing delegation, convert it to valset and store it.
+// 3. if valset exists, but user doesnot have existing delegation, modify current valset.
+// 4. if valset exists & user has existing delegation, modify current valset.
```

Comment:
note: Really want to make sure i'm getting the logic right here

Reply:
Wait I'm confused, if you call set, you should be inputting a validator set?

So only decision criterion is "if set validator list provides empty list or not". If its an empty-list, then copy live delegation distribution, otherwise make a new valset

Reply:
shouldn't it be "if setValList provides empty && check if user doesnot have existing valset && check if user has existing delegation" then copy live delegation distribution?

Reply:
or do you think it's a better decision to have another message that 'ConvertsExistingDelegationsToValSet` ?